### PR TITLE
fix: TT-29 Clean the form if the customer is removed

### PR DESCRIPTION
**Pull request by LEON12699**

### CODE REVIEW

File: src/app/modules/customer-management/components/customer-info/components/customer-list/customer-list.component.ts [hunk 1]
```diff
@@ -109,6 +121,10 @@ export class CustomerListComponent implements OnInit, OnDestroy, AfterViewInit {
   }
 
   deleteCustomer() {
+    if (this.idToDelete === this.idToEdit) {
```
Comment (scastillo-jp) [inline src/app/modules/customer-management/components/customer-info/components/customer-list/customer-list.component.ts:124]: I think it is necessary to check these conditions in the following cases:
- When select te project **A** to edit and clicked in other project **B** options Delete 🗑.
- when select  the project  **A** with the option edit and click in delete 🗑.

File: src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.ts [hunk 2]
```diff
@@ -81,6 +80,7 @@ export class CreateCustomerComponent implements OnInit, OnDestroy {
       this.markTabsAsInactive();
       this.customerForm.reset();
     }
+    this.hasChangedEvent.emit((this.hasChange = false));
```
[the remark this answers is not in the document]
Comment (LEON12699) [reply src/app/modules/customer-management/components/customer-info/components/create-customer/create-customer.ts:83]: All of this criterias of acceptance ... Passed  ✅ 
But I found another criteria that I think fails in the actual implementation. 

- When edit the tab: customer information (customer TEST) and select the same customer (customer Test)  to edit A modal alert should appear with a message and when selecting the confirm button the modal closes, discards the current changes and loads the customer data.  (actual not discard changes)

 @scastillo-jp  it is a fail or not ?  I can work on it in this PR if it is

File: src/app/modules/customer-management/components/customer-info/components/customer-list/customer-list.component.ts (hunk 1 above)
Comment (LEON12699) [reply src/app/modules/customer-management/components/customer-info/components/customer-list/customer-list.component.ts]: - In the first case the data of project **A** is kept in the form and project **b** is deleted. 
- In the second case the form is reset and project **A** is deleted.

File: src/app/modules/customer-management/components/customer-info/components/customer-list/customer-list.component.ts [hunk 3]
```diff
@@ -109,6 +125,10 @@ export class CustomerListComponent implements OnInit, OnDestroy, AfterViewInit {
   }
 
   deleteCustomer() {
+    if (this.idToDelete === this.currentCustomerIdToEdit) {
+      this.store.dispatch(new ResetCustomerToEdit());
+      this.resetProjectFieldsToEdit();
+    }
```
Comment (thegreatyamori) [inline src/app/modules/customer-management/components/customer-info/components/customer-list/customer-list.component.ts:131]: I think that is code sequence in line 114 is exactly to this by the exception `this.resetProjectFieldsToEdit();` & `this.idToEdit`
```typescript
if (this.currentCustomerIdToEdit === this.idToEdit) {
  this.store.dispatch(new ResetCustomerToEdit());
}
```
Is there any way to extract them to a method?

**Review by thegreatyamori — COMMENTED** [review]
Great Job !

**Review by scastillo-jp — APPROVED** [review]

**Pull request merged**